src/main/java/org/sagebionetworks/bridge/services/ActivityEventService.java

```diff
@@ -147,9 +165,58 @@ public void publishActivityEvent(ActivityEvent event) {
      */
     public Map<String, DateTime> getActivityEventMap(String healthCode) {
         checkNotNull(healthCode);
+        
         return activityEventDao.getActivityEventMap(healthCode);
     }
 
+    /**
+     * Gets the activity events times for a specific user in order to schedule against them.
+     */
+    public Map<String, DateTime> getActivityEventMap(Account account) {
+        checkNotNull(account.getHealthCode());
+        Map<String, DateTime> activityMap = activityEventDao.getActivityEventMap(account.getHealthCode());
+        Builder<String, DateTime> builder = ImmutableMap.<String, DateTime>builder();
+        
+        DateTime studyStartDateTime = activityMap.get(ActivityEventObjectType.STUDY_START_DATE.name().toLowerCase());
+        DateTime activitiesRetrievedDateTime = activityMap.get(ActivityEventObjectType.ACTIVITIES_RETRIEVED.name().toLowerCase());
+        DateTime enrollmentDateTime = activityMap.get(ActivityEventObjectType.ENROLLMENT.name().toLowerCase());
+        
+        if (activitiesRetrievedDateTime != null) {
+            studyStartDateTime = activitiesRetrievedDateTime;
+        } else if (enrollmentDateTime != null) {
+            studyStartDateTime = enrollmentDateTime;
+        } else if (studyStartDateTime == null) {
+            Study study = studyService.getStudy(account.getStudyId());
+            StudyParticipant studyParticipant = participantService.getParticipant(study, account.getHealthCode());
+            studyStartDateTime = studyParticipant.getCreatedOn();
+            builder.put(ActivityEventObjectType.STUDY_START_DATE.name().toLowerCase(), studyStartDateTime);
+        }
+        
+        for (Map.Entry<String, DateTime> entry : activityMap.entrySet()) {
+            builder.put(entry.getKey(), entry.getValue());
+        }
+        
+        return builder.build();
+    }
+    
+    public List<ActivityEvent> getActivityEventList(Account account) {
+        Map<String, DateTime> activityEvents = getActivityEventMap(account);
+
+        List<ActivityEvent> activityEventList = Lists.newArrayList();
+        for (Map.Entry<String, DateTime> entry : activityEvents.entrySet()) {
+            DynamoActivityEvent event = new DynamoActivityEvent();
+            event.setEventId(entry.getKey());
+
+            DateTime timestamp = entry.getValue();
+            if (timestamp !=null) {
+                event.setTimestamp(timestamp.getMillis());
+            }
+
+            activityEventList.add(event);
+        }
+        return activityEventList;
+    }
```

Comment:
The problem with this approach is that the existing methods that take healthCode will now return different results than the methods that take an account, which is unnecessarily confusing. What about removing the path through ParticipantService entirely, and adding the AccountDao to the ActivityEventService? If you need the account, you can load it from the methods that take a health code (you don't always need it, as we discussed).